### PR TITLE
fix(video.js): missing 'id' property for player/component creation

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- https://github.com/videojs/video.js/issues/5088

Thanks!

Fixe #59436

/cc @BrainCrumbz

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

### CODE REVIEW

File: types/video.js/index.d.ts [hunk 1]
```diff
@@ -1986,9 +1986,10 @@ declare namespace videojs {
     };
 
     interface ComponentOptions {
-        children?: undefined | Child[];
-        createEl?: boolean;
-        el?: HTMLElement;
+        children?: Child[] | undefined;
+        createEl?: boolean | undefined;
+        el?: HTMLElement | undefined;
+        id?: string | undefined;
```
Comment (joeflateau) [inline types/video.js/index.d.ts:1992]: given that this property is undocumented on video.js I don't believe it belongs in type definitions. they mention in the other issue (https://github.com/videojs/video.js/issues/7670) that this property SHOULD be documented but it isn't yet. my vote is that this does *not yet* belong here.

**Review by joeflateau — CHANGES_REQUESTED** [review]

**Comment by peterblazejewicz** [comment]
closing as per @joeflateau last comment

**Comment by BrainCrumbz** [comment]
@peterblazejewicz @joeflateau just to mention that property is currently documented. 
See:
* https://github.com/videojs/videojs.com/pull/139
* https://videojs.com/guides/options/#id

**Comment by peterblazejewicz** [comment]
@BrainCrumbz thx!

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## video.js ([<kbd>unpkg</kbd>](https://unpkg.com/browse/video.js@latest/))
was missing the following properties:
1. The declaration doesn't match the JavaScript module 'video.js'. Reason:
The JavaScript module can be called or constructed, but the declaration module cannot.

The most common way to resolve this error is to use 'export =' syntax.
To learn more about 'export =' syntax, see https://www.typescriptlang.org/docs/handbook/modules.html#export--and-import--require.
1. hooks_
1. hooks
1. hook
1. hookOnce

<details>
<summary>as well as these 52 other properties...</summary>
<p>removeHook, VERSION, options, getPlayers, getPlayer, getAllPlayers, players, getComponent, registerComponent, getTech, registerTech, use, middleware, browser, TOUCH_ENABLED, extend, mergeOptions, bind, registerPlugin, deregisterPlugin, plugin, getPlugins, getPlugin, getPluginVersion, addLanguage, log, createLogger, createTimeRange, createTimeRanges, formatTime, setFormatTime, resetFormatTime, parseUrl, isCrossOrigin, EventTarget, on, one, off, trigger, xhr, TextTrack, AudioTrack, VideoTrack, computedStyle, dom, url, mediaSources, MediaSource, URL, HlsHandler, HlsSourceHandler, Hls</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 0ea382b8d1a3367e6932c7f7a61da80ce921f22a
</p>

**Review by joeflateau — APPROVED** [review]
thank you

**Comment by peterblazejewicz** [comment]
Ready to merge 💖

**Pull request merged**